transports/tls-quic/src/certificate.rs

```diff
@@ -275,6 +269,157 @@ impl P2pCertificate<'_> {
         // if signature verification fails.
         self.extension.public_key.verify(&msg, &self.extension.signature)
     }
+
+    /// Return the signature scheme corresponding to [`AlgorithmIdentifier`]s
+    /// of `subject_pki` and `signature_algorithm`
+    /// according to https://tools.ietf.org/id/draft-ietf-tls-tls13-21.html#rfc.section.4.2.3.
+    pub fn signature_scheme(&self) -> Option<rustls::SignatureScheme> {
+        // Certificates MUST use the NamedCurve encoding for elliptic curve parameters.
+        // Endpoints MUST abort the connection attempt if it is not used.
+        use rustls::SignatureScheme::*;
+        use oid_registry::*;
+        let signature_algorithm = &self.certificate.signature_algorithm;
+        let pki_algorithm = &self.certificate.tbs_certificate.subject_pki.algorithm;
+
+        if pki_algorithm.algorithm == OID_PKCS1_RSAENCRYPTION {
+            if signature_algorithm.algorithm == OID_PKCS1_SHA256WITHRSA {
```

Comment:
I'd rather replace this sequence with `match`

Reply:
It wont make the code look better because of https://stackoverflow.com/questions/28225958/why-is-this-match-pattern-unreachable-when-using-non-literal-patterns.